include/cantera/kinetics/Reaction.h

```diff
@@ -502,6 +452,149 @@ class BlowersMaselInterfaceReaction : public BlowersMaselReaction
     std::string sticking_species;
 };
 
+
+//! An intermediate class used to avoid naming conflicts of 'rate' member
+//! variables and getters (see `ElementaryReaction`, `PlogReaction` and
+//! `ChebyshevReaction`).
+class Reaction3 : public Reaction
+{
+public:
+    Reaction3() : Reaction() {}
+    Reaction3(const Composition& reactants, const Composition& products);
+
+    //! Get reaction rate pointer
+    shared_ptr<ReactionRateBase> rate() {
+        return m_rate;
+    }
```

Comment:
I honestly believe that much of what of the refactoring is doing here will remain 'invisible' to the grand majority of users. If you query reactions, you may get an occasional '3' in the end, which imho is beneficial for debugging purposes. For the few users that explicitly use the C++ interface, there are breaking changes, so having a clear deprecation pathway with separate `*Reaction` and `*Reaction3` objects makes things very clear.

Reply:
To update this response here: introducing `Reaction3` here was deliberate to clearly illustrate what’s different. I don’t think that the additional inheritance layer is performance limiting, and it may be instructive for those who work ‘under the hood’, especially when transitioning code. I am, however, not entirely opposed to adding the new functions to the base object (which thus far was largely unchanged).

Reply:
Ok. As member variables `rate` and member function `rate()` are not mutually exclusive, I removed `Reaction3`.